src/coreclr/jit/fgbasic.cpp

```diff
@@ -955,35 +1000,126 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
+                        {
+                            if (FgStack::IsConstant(pushedStack.Pop()))
+                            {
+                                foldableIntrinsc = true;
+                                pushedStack.PushConstant();
+                            }
+                            else
+                            {
+                                pushedStack.PushUnknown();
+                            }
+                            break;
+                        }
+
+                        //// the following needs https://github.com/dotnet/runtime/pull/52156
+                        // These are foldable if two arguments are constants
+                        //case NI_System_Type_op_Equality:
+                        //case NI_System_Type_op_Inequality:
+                        //case NI_System_String_get_Chars:
+                        case NI_System_Type_IsAssignableTo:
+                        case NI_System_Type_IsAssignableFrom:
+                        {
+                            if (FgStack::IsConstant(pushedStack.Pop()) && FgStack::IsConstant(pushedStack.Pop()))
+                            {
+                                foldableIntrinsc = true;
+                                pushedStack.PushConstant();
+                            }
+                            else
+                            {
+                                pushedStack.PushUnknown();
+                            }
+                        }
+
                         case NI_IsSupported_True:
                         case NI_IsSupported_False:
                         {
+                            foldableIntrinsc = true;
                             pushedStack.PushConstant();
                             break;
                         }
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+                        case NI_Vector128_get_Count:
+                        case NI_Vector256_get_Count:
+                            foldableIntrinsc = true;
+                            pushedStack.PushConstant();
+                            break;
+                        case NI_Vector256_get_Zero:
+                        case NI_Vector256_get_AllBitsSet:
+                            foldableIntrinsc = true;
+                            pushedStack.PushUnknown();
+                            break;
+#elif defined(TARGET_ARM64) && defined(FEATURE_HW_INTRINSICS)
+                        case NI_Vector64_get_Count:
+                        case NI_Vector64_get_Count:
+                            foldableIntrinsc = true;
+                            pushedStack.PushConstant();
+                            break;
+                        case NI_Vector128_get_Zero:
+                        case NI_Vector128_get_AllBitsSet:
+                            foldableIntrinsc = true;
+                            pushedStack.PushUnknown();
+                            break;
+#endif
```

Comment:
Create is another interesting case, currently if all arguments are constant we'll fold it to a constant.

In the future we would ideally also do folding if "most" inputs are constant.

Reply:
Good catch, for now I'll leave it as "if first arg is a const - we most likely will fold it" - I mean how popular patterns such as `Vector.Create(cns, x, x2, ...)` anyway

Reply:
I thought I had suggested this, but we should consider refactoring this method so it's not so large. Likely it would improve jit throughput and make things more understandable.

Consider this as a follow-up activity, since it's best to verify changes like this don't cause diffs.